src/test/java/com/google/research/bleth/simulator/GlobalResolverTest.java

```diff
@@ -0,0 +1,254 @@
+        assertThat(resolver.getBoard().getAgentsOnLocation(TWO_ON_TWO_COORDINATE)).isEmpty();
+    }
+
+    @Test
+    public void secondEstimateWithoutInformation_DoNotUpdateTheBoard() {
+        RealBoard realboard = new RealBoard(3, 3);
+        Beacon beacon = createRandomBeaconOnLocation(ZERO_ON_ZERO_COORDINATE, realboard);
+        GlobalResolver resolver = GlobalResolver.createResolver(/* rowsNum= */ 3,
+                                                                /* colsNum= */ 3,
+                                                                ImmutableList.of(beacon));
+        resolver.receiveInformation(TWO_ON_TWO_COORDINATE, ImmutableList.of(beacon.transmit()));
+        resolver.estimate();
+
+        resolver.estimate();
+
+        assertThat(resolver.getBoard().getAgentsOnLocation(TWO_ON_TWO_COORDINATE)).containsExactly(beacon);
+    }
+
+    @Test
+    public void secondEstimateForUnlocatedBeacon_UpdateAccordingToTheObserverLocationOnly() {
+        RealBoard realboard = new RealBoard(3, 3);
+        Beacon beacon = createRandomBeaconOnLocation(ZERO_ON_ZERO_COORDINATE, realboard);
+        GlobalResolver resolver = GlobalResolver.createResolver(/* rowsNum= */ 3,
+                                                                /* colsNum= */ 3,
+                                                                ImmutableList.of(beacon));
+        resolver.estimate();
+
+        resolver.receiveInformation(TWO_ON_TWO_COORDINATE, ImmutableList.of(beacon.transmit()));
+        resolver.estimate();
+
+        assertThat(resolver.getBoard().getAgentsOnLocation(TWO_ON_TWO_COORDINATE)).containsExactly(beacon);
+    }
+
+    @Test
+    public void transmissionsFromPreviousRoundsIsDeleted() {
+        RealBoard realboard = new RealBoard(5, 5);
+        Beacon beacon = createRandomBeaconOnLocation(ONE_ON_ZERO_COORDINATE, realboard);
+        GlobalResolver resolver = GlobalResolver.createResolver(/* rowsNum= */ 5,
+                                                                /* colsNum= */ 5,
+                                                                ImmutableList.of(beacon));
+        resolver.receiveInformation(ZERO_ON_ZERO_COORDINATE, ImmutableList.of(beacon.transmit()));
+        resolver.receiveInformation(ZERO_ON_ZERO_COORDINATE, ImmutableList.of(beacon.transmit()));
+        resolver.receiveInformation(ZERO_ON_ZERO_COORDINATE, ImmutableList.of(beacon.transmit()));
+        resolver.estimate();
+
+        resolver.receiveInformation(new Location(4, 4), ImmutableList.of(beacon.transmit()));
+        resolver.estimate();
+
+        assertThat(resolver.getBoard().getAgentsOnLocation(TWO_ON_TWO_COORDINATE)).containsExactly(beacon);
+    }
+
+    private Beacon createRandomBeaconOnLocation(Location initialLocation, RealBoard realBoard) {
+        return BEACON_FACTORY.createBeacon(initialLocation, new RandomMovementStrategy(), realBoard);
+    }
+
+    private boolean isBoardEmpty(Board board, int rowsNum, int colsNum) {
```

Comment:
Doesn't a Board provide methods for getting its size directly? (Wouldn't having such methods simplify all callers that currently need to supply the size with the Board whenever they pass it somewhere else?)